File: app/local_plugins/assistant_server/android/src/main/kotlin/com/aiagent/assistant_server/AssistantEvents.kt
package com.aiagent.assistant_server

/**
 * assistant_server → Flutter EventChannel 的事件 payload 工厂。
 *
 * 所有事件统一带 `type` 字段。Flutter 端按 `type` 分派：
 *  - `message`        : 对话消息，必带 `role` (user/assistant) + `stage` (partial/final) + `text`
 *  - `sessionState`   : `state` ∈ starting / active / stopping / stopped / error
 *  - `error`          : `code` + `message` + 可选 `role`
 *  - `connectionState`: agent 端到端服务连接状态
 */
internal object AssistantEvents {

    fun message(
        sessionId: String,
        role: String,
        stage: String,
        text: String,
        requestId: String? = null,
    ): Map<String, Any?> = mapOf(
        "type" to "message",
        "sessionId" to sessionId,
        "role" to role,
        "stage" to stage,
        "text" to text,
        "requestId" to requestId,
    )

    fun sessionState(sessionId: String, state: String, errorMessage: String? = null): Map<String, Any?> =
        mapOf(
            "type" to "sessionState",
            "sessionId" to sessionId,
            "state" to state,
            "errorMessage" to errorMessage,
        )

    fun error(
        sessionId: String,
        code: String,
        message: String,
        role: String? = null,
        fatal: Boolean = false,
    ): Map<String, Any?> = mapOf(
        "type" to "error",
        "sessionId" to sessionId,
        "code" to code,
        "message" to message,
        "role" to role,
        "fatal" to fatal,
    )

    fun connectionState(
        sessionId: String,
        state: String,
        errorMessage: String? = null,
    ): Map<String, Any?> = mapOf(
        "type" to "connectionState",
        "sessionId" to sessionId,
        "state" to state,
        "errorMessage" to errorMessage,
    )
}

/**
 * AI 助理的对话角色。user = 戴耳机的本机用户；assistant = AI 回复。
 */
internal enum class AssistantRole(val wireName: String) {
    USER("user"),
    ASSISTANT("assistant"),
}

File: apps/client/local_plugins/assistant_server/ios/Classes/AssistantEvents.swift
import Foundation

/// assistant_server → Flutter EventChannel 的事件 payload 工厂。
///
/// 所有事件统一带 `type` 字段。Flutter 端按 `type` 分派：
///  - `message`        : 对话消息，必带 `role` (user/assistant) + `stage` (partial/final) + `text`
///  - `sessionState`   : `state` ∈ starting / active / stopping / stopped / error
///  - `error`          : `code` + `message` + 可选 `role`
///  - `connectionState`: agent 端到端服务连接状态
enum AssistantEvents {

    static func message(
        sessionId: String, role: String, stage: String,
        text: String, requestId: String? = nil
    ) -> [String: Any?] {
        return [
            "type": "message",
            "sessionId": sessionId,
            "role": role,
            "stage": stage,
            "text": text,
            "requestId": requestId,
        ]
    }

    static func sessionState(
        sessionId: String, state: String, errorMessage: String? = nil
    ) -> [String: Any?] {
        return [
            "type": "sessionState",
            "sessionId": sessionId,
            "state": state,
            "errorMessage": errorMessage,
        ]
    }

    static func error(
        sessionId: String, code: String, message: String,
        role: String? = nil, fatal: Bool = false
    ) -> [String: Any?] {
        return [
            "type": "error",
            "sessionId": sessionId,
            "code": code,
            "message": message,
            "role": role,
            "fatal": fatal,
        ]
    }

    static func connectionState(
        sessionId: String, state: String, errorMessage: String? = nil
    ) -> [String: Any?] {
        return [
            "type": "connectionState",
            "sessionId": sessionId,
            "state": state,
            "errorMessage": errorMessage,
        ]
    }
}

/// AI 助理的对话角色。user = 戴耳机的本机用户；assistant = AI 回复。
enum AssistantRole: String {
    case user = "user"
    case assistant = "assistant"
}
